zap/src/main/java/org/parosproxy/paros/core/scanner/AbstractPlugin.java

```diff
@@ -650,6 +674,38 @@ protected boolean isPageOther(HttpMessage msg) {
         return isCustomPage(msg, CustomPage.Type.OTHER);
     }
 
+    /**
+     * Tells whether or not the response has a status code between 400 and 499 (inclusive). Falls
+     * back to check {@code CustomPage.Type.NOTFOUND_404}. Checks if the message matches {@code
+     * CustomPage.Type.OK_200} or {@code CustomPage.Type.ERROR_500} first, in case the user is
+     * trying to override something.
+     *
+     * @param msg the message that will be checked
+     * @return {@code true} if the message matches, {@code false} otherwise
+     * @since TODO Add version
+     */
+    public boolean isClientError(HttpMessage msg) {
+        return isPage404(msg)
```

Comment:
This would have to be inlined? `isPage404` might return false if page 200 match, so we should not be checking the status code in that case?

Reply:
So status code will never be checked? Or:
```java
public boolean isClientError(HttpMessage msg) {
    if (isCustomPage(msg, CusotmPage.Type.OK_200) || isCUstomPage(msg, CustomPage.Type.ERROR_500) {
        return false;
    }
    return isCustomPage(msg, CustomPage.Type.NOTFOUND_404)
            ? true
            : HttpStatusCode.isClientError(msg.getResponseHeader().getStatusCode());
}
```

Reply:
Yes, the latter.